src/UrlGenerator.php

```diff
@@ -0,0 +1,332 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Router\FastRoute;
+
+use Yiisoft\Router\Group;
+use Yiisoft\Router\Route;
+use Yiisoft\Router\RouteCollectorInterface;
+use Yiisoft\Router\RouteNotFoundException;
+use Yiisoft\Router\UrlMatcherInterface;
+use Yiisoft\Router\UrlGeneratorInterface;
+use FastRoute\RouteParser;
+
+use function array_key_exists;
+use function array_keys;
+use function implode;
+use function is_string;
+use function preg_match;
+
+class UrlGenerator implements UrlGeneratorInterface
+{
+    /** @var string */
+    private $uriPrefix = '';
+
+    /**
+     * All attached routes as Route instances
+     *
+     * @var Route[]
+     */
+    private $routes = [];
+
+    /**
+     * @var UrlMatcherInterface $matcher
+     */
+    private $matcher;
+
+    /**
+     * @var RouteCollectorInterface $collector
+     */
+    private $collector;
+
+    /**
+     * @var RouteParser
+     */
+    private $routeParser;
+
+    /**
+     * Constructor
+     *
+     * @param UrlMatcherInterface $matcher url matcher
+     * @param RouteCollectorInterface $collector route collector
+     */
+    public function __construct(
+        UrlMatcherInterface $matcher,
+        RouteCollectorInterface $collector
+    ) {
+        $this->matcher = $matcher;
+        $this->collector = $collector;
+        $this->routeParser = new RouteParser\Std();
```

Comment:
Should be passed as dependency

Reply:
> Should be passed as dependency

For what purpose?

Reply:
So that it can be configured, like it was before

Reply:
> So that it can be configured, like it was before

No, it can't, it is an adapter.

Reply:
I've thought maybe this makes sense.